File: Kotlin/problem004.kt
/*
 * Problem 4 - Project Euler
 * http://projecteuler.net/index.php?section=problems&id=4
 */
fun isPalindromic(x: Int): Boolean {
    return x.toString().reversed() == x.toString()
}
fun main(args: Array<String>) {
    val pmax = (100..1000).map { i ->
                (i..1000).map { j -> i * j }
            }
            .flatten()
            .distinct()
            .filter { isPalindromic(it) }
            .max()

    println(pmax)
}

File: Swift/problem004.swift
// Problem 4 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=4
func is_palindromic(x: Int) -> Bool {
    return String(String(x).reversed()) == String(x)
}

var maxnum = 0
for i in 100...999 {
    for j in i...999 {
        let n = i * j
        if is_palindromic(x: n) {
            maxnum = max(maxnum, n)
        }
    }
}
print(maxnum)
